### PR TITLE
Give assigned tickets a time to live, default 10 minutes

### PR DESCRIPTION
**Pull request by Laremere**
Resolves https://github.com/googleforgames/open-match/issues/1168

### CODE REVIEW

File: install/helm/open-match/templates/om-configmap-override.yaml [hunk 1]
```diff
@@ -33,4 +33,5 @@ data:
     synchronizer:
       registrationIntervalMs:  {{ index .Values "open-match-core" "registrationIntervalMs" }}
       proposalCollectionIntervalMs: {{ index .Values "open-match-core" "proposalCollectionIntervalMs" }}
+    assignedDeleteTimeout: {{ index .Values "open-match-core" "proposalCollectionIntervalMs" }}
```
Comment (sawagh) [inline install/helm/open-match/templates/om-configmap-override.yaml:36]: you mean assignedDeleteTimeout?

File: install/helm/open-match/values-production.yaml [hunk 2]
```diff
@@ -173,6 +173,7 @@ redis:
 open-match-core:
   enabled: true
   ignoreListTTL: 60000ms
```
Comment (sawagh) [inline install/helm/open-match/values-production.yaml:175]: may be represent everything in excess of a minute as a minutes value across the yamls? (like you did for this variable in the values.yaml file?

File: internal/statestore/redis.go [hunk 3]
```diff
@@ -529,13 +534,36 @@ func (rb *redisBackend) UpdateAssignments(ctx context.Context, req *pb.AssignTic
...
 		}
 
-		cmds = append(cmds, ticket.GetId(), ticketByte)
+		err = redisConn.Send("SET", ticket.Id, ticketByte, "PX", int64(assignmentTimeout), "XX")
+		if err != nil {
+			return nil, errors.Wrap(err, "error sending ticket assignment set")
+		}
 	}
 
-	_, err = redisConn.Do("MSET", cmds...)
```
Comment (sawagh) [inline internal/statestore/redis.go:535]: Wasnt moving to MSET an optimization? Why move back to EXEC? and if there are reasons, then may be do that consistently for other patterns in the storage layer?

https://github.com/googleforgames/open-match/commit/3de052279b5a725c2980851a29056b8a45ead401

Comment (Laremere) [reply internal/statestore/redis.go:535]: Redis does not provide a function which both sets multiple values, and also allows setting the expiration at the same time.  As far as I can see there's only setting expiration one ticket at a time.  So it's fewer commands to set them with their expiration.  As a bonus, it's possible to say only set they key if it exists, which prevents a race condition if the ticket is deleted between reading and writing.

File: install/helm/open-match/templates/om-configmap-override.yaml (hunk 1 above)
Comment (Laremere) [reply install/helm/open-match/templates/om-configmap-override.yaml:36]: great catch!

File: install/helm/open-match/values-production.yaml (hunk 2 above)
Comment (Laremere) [reply install/helm/open-match/values-production.yaml:175]: Yeah, they should be in sync. Fixed.

**Pull request merged**